packages/tangle/ledgerstate.go

```diff
@@ -194,17 +194,15 @@ func (l *LedgerState) SnapshotUTXO() (snapshot *ledgerstate.Snapshot) {
 		unspentOutputs := make([]bool, len(transaction.Essence().Outputs()))
 		includeTransaction := false
 		for i, output := range transaction.Essence().Outputs() {
-			l.CachedOutputMetadata(output.ID()).Consume(func(outputMetadata *ledgerstate.OutputMetadata) {
-				if outputMetadata.ConfirmedConsumer() == ledgerstate.GenesisTransactionID { // no consumer yet
-					unspentOutputs[i] = true
-					includeTransaction = true
-				} else {
-					tx, exist := copyLedgerState[outputMetadata.ConfirmedConsumer()]
-					// ignore consumers that are not confirmed long enough or even in the future.
-					if !exist || startSnapshot.Sub(tx.Essence().Timestamp()) < minAge {
-						unspentOutputs[i] = true
-						includeTransaction = true
-					}
+			unspentOutputs[i] = true
+			includeTransaction = true
+
+			l.ConfirmedConsumer(output.ID()).Consume(func(consumer *ledgerstate.Consumer) {
```

Comment:
If I read `ConfirmedConsumer()` right, it could be that the `*ConfirmedConsumer` returned is nil if there is none, in which case this `Consume` would cause a panic?